src/context.cpp

```diff
@@ -379,6 +379,27 @@ namespace librealsense
     }
 
 #ifdef BUILD_WITH_DDS
+    struct sid_index
+    {
+        int16_t sid;   // Stream ID; assigned based on the stream TYPE, really
+        int8_t index;  // Used to distinguish similar streams like IR L / R, 0 otherwise
+
+        sid_index( int sid_, int index_ )
+            : sid( static_cast< int16_t >( sid_ ) )
+            , index( static_cast< int8_t >( index_ ) )
+        {
+        }
+
+        sid_index() = default;
+        sid_index( sid_index const& ) = default;
+        sid_index( sid_index&& ) = default;
+
+        std::string to_string() const { return '(' + std::to_string( sid ) + '.' + std::to_string( index ) + ')'; }
+
+        inline bool operator<( sid_index const & r ) const { return this->sid < r.sid  ||  this->index < r.index; }
```

Comment:
I don't think this is a well defined "smaller than" operation. If a = { 5, 0 }  and b = { 3, 1 } then a < b is true and also b < a is true;

Reply:
It doesn't matter, as long as it's consistent. This is simply a requirement for ordering items in a map.
That said, I think it's logical to break by `sid` and then `index` -- which is what I did, no?

Reply:
It is logical to break `sid` and then `index`, but I think it means:
`this->sid < r.sid || ( this->sid == r.sid && this->index < r.index )`
No case when `this->sid > r.sid` will return `true`

Reply:
Right :)